### DISCUSSION TITLE
Gerar PDF com loop

### DISCUSSION THREAD

**Discussion by rios-se** [dompdf/dompdf#2464] [General]
Olá, gostaria de saber se é possível criar um loop para gerar várias páginas iguais. desde já agradeço.
[>>Código completo<<][1]Preciso gerar um pdf para cada cpf, digitando vários cpfs de uma única vez, porem, o pdf traz apenas uma única página com o cpf, em vez de várias paginas, cada uma com um cpf.

```
<?php

use Dompdf\Dompdf;

require_once 'dompdf/autoload.inc.php';
$datatual = date('d/m/Y');
$valorEntrada = $_GET["cpf"];
$valTratado = explode(" ", $valorEntrada);
$contCpf = count($valTratado);
for ($i = 0; $i < $contCpf; $i++) {
    $dompdf = new DOMPDF();
    $html = "<p> Teste de CPF: $valTratado[$i] <p/>";
    $dompdf->loadHtml($html);
    $dompdf->render();
    $dompdf->stream("Documentação em lote.pdf",
        array("Attachment" => false));
    $dompdf->output();
}
?>
```

  [1]: https://uploaddeimagens.com.br/imagens/_qTRWhM

**Comment by bsweeney** [comment]
The primary issue is that you are streaming the PDF, which sends the contents of the rendered PDF to the browser. After you do that the first time you won't really be able to send anything else to the user. Are you wanting to generate multiple PDFs and send them all to the user? Or generate a single PDF that has the same page repeated for each iteration?

---

O principal problema é que você está transmitindo o PDF, que envia o conteúdo do PDF renderizado para o navegador. Depois de fazer isso pela primeira vez, você não conseguirá enviar mais nada para o usuário. Você deseja gerar vários PDFs e enviá-los todos ao usuário? Ou gerar um único PDF com a mesma página repetida para cada iteração?
(via Google Translate)

**Comment by rios-se** [comment]
Desejo gerar vários pdfs, e salvar cada pdf em sua devida pasta.
exemplo, tenho 10 clientes, e quero gerar um pdf para cada um deles, e salvar em um pasta com o nome de cada um deles

**Reply by bsweeney** [reply]
Try something like:

```
use Dompdf\Dompdf;
require_once 'dompdf/autoload.inc.php';
$datatual = date('d/m/Y');
$valorEntrada = $_GET["cpf"];
$valTratado = explode(" ", $valorEntrada);
$contCpf = count($valTratado);
for ($i = 0; $i < $contCpf; $i++) {
    $dompdf = new DOMPDF();
    $html = "<p> Teste de CPF: $valTratado[$i] <p/>";
    $dompdf->loadHtml($html);
    $dompdf->render();
    $pdf = $dompdf->output();
    file_put_contents("/path/to/user".$i."filename.pdf", $pdf);
}
?>
```
